documentation/writing_tests.md

```diff
@@ -422,6 +424,77 @@ rush add --dev -p @azure/test-utils-recorder
 
+
+
+## Engineering setup
+
+While writing tests for the Azure SDK works both as a way to verify that our code is correct, and a way to share how to use our code with our customers, we must keep in mind that tests also triggered by automatic systems that generate nighly builds of our packages, that verify that our commits are correct, and that help us have a better level of confidence before releasing anything to the public. 
+
+For our Egineering Systems to pick up our tests appropriately, our packages must be configured according to their guidelines, and our tests must be thought to address their needs. In this section we will go through some of these concepts, and provide links that expand them in detail. We will be covering:
+
+- [Engineering goals](#engineering-goals).
+- [CI configuration files](#ci-configuration-files).
+- [Activating the pipeline](#activating-the-pipeline).
+- [Delivering live tests to our users](#delivering-live-tests-to-our-users).
+
+### Engineering goals
+
+Though the tests for the Azure SDK for JavaScript and TypeScript must target live resources, we should make sure they only do so when necessary. For this purpose, we must keep in mind the following guidelines:
+
+- Tests should not be flaky. Tests should pass regardless of who's executing it, when it is running, and how many times does it run.
+- While writing tests, use static resources for your tests when possible to avoid repeatedly incurring delays for setting up and tearing down resources. Once tests are pushed into the repository, the pipelines will create these resources for us. We'll examine how to set this up in the following sections.
+- Avoid calling to timed delays (like `setTimeout`) to assert that a change happened in the live resources. Also avoid locking the main thread until the resource responds.
+- The resources created in the tests should be unique. Running the same test in parallel, multiple times, should not break them.
+
+You can read more recomendations through the following link: [Best Practices for writing tests that target live resources](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/51/Testing-Guidelines).
+
+### CI configuration files
+
+To ensure that our tests are executed in the test [Azure DevOps pipelines](https://azure.microsoft.com/en-us/services/devops/pipelines/) that have been previously configured by our team, some configuration files are necessary:
+
+- A file named `ci.yml` will be automatically generated at the common parent of the clients of a specific service (let's say, at the `keyvault/` level of `keyvault/keyvault-keys`).
+- A `test-resources.json` file, also at the common parent of the clients of a service, which will contain an [ARM template](https://docs.microsoft.com/en-us/azure/azure-resource-manager/templates/overview) of the resources needed to run all of the tests of these clients.
```

Comment:
These can also be organized into the package folders (or somewhere else in the hierarchy). New-TestResources.ps1 will search for these files within the service directory. https://github.com/Azure/azure-sdk-tools/blob/master/eng/common/TestResources/New-TestResources.ps1.md#description ... At this time all will be deployed because we don't have a way to determine which individual package is being tested across all languages.